test/gtest/ucp/test_ucp_am.cc

```diff
@@ -692,6 +692,111 @@ class test_ucp_am_nbx : public test_ucp_am_base {
     ucp_mem_h                       m_rx_memh;
 };
 
+class test_ucp_am_id : public test_ucp_am_nbx {
+public:
+    test_ucp_am_id()
+    {
+        reset_counters();
+    }
+
+protected:
+
+    void reset_counters()
+    {
+        test_ucp_am_nbx::reset_counters();
+        m_recv_counter_cb_1 = 0;
+        m_recv_counter_cb_2 = 0;
+    }
+
+    ucs_status_t set_am_id_overflow_data_handler(entity &e, unsigned am_id,
```

Comment:
It's different in that `set_am_data_handler` receives `am_id` as `uint16_t` instead of `unsigned int`, and it also asserts successful registration of data handler while `set_am_id_overflow_data_handler` only returns status, so the opposite case can also be tested when using this function. I can change its name though as there is no overflow here.. or maybe I can change `set_am_data_handler` so it would be more generic.
What do you think?

Reply:
I see.
Than maybe have `set_am_data_handler` in the base class, which would call `set_am_data_handler_internal` - which will do the job and return rc.
`set_am_data_handler` will `ASSERT_...` so that no change in existing code is needed.

Your new func `set_am_..something` in the subclass will call `set_am_data_handler_internal` as well.

Reply:
Oh, I missed this comment yesterday. I can revert it